matminer/data_retrieval/retrieve_MP.py

```diff
@@ -24,19 +25,39 @@ def __init__(self, api_key=None):
     def api_link(self):
         return "https://materialsproject.org/wiki/index.php/The_Materials_API"
 
-    def get_dataframe(self, criteria, properties, index_mpid=True, **kwargs):
+    def get_dataframe(self, criteria, properties, index_mpid=True,
+                      line_mode=True, **kwargs):
         """
         Gets data from MP in a dataframe format. See api_link for more details.
 
         Args:
-            all arguments including criteria, properties and index_mpid are the
-            same as in get_data
+            criteria (dict): the same as in get_data
+            properties ([str]): the same properties supported as in get_data
+                plus: "structure", "initial_structure", "final_structure",
+                "bandstructure", "phonon_bandstructure", "phonon_ddb",
+                "phonon_bandstructure", "phonon_dos". Note that for a long list
+                of compounds, it make take a long time to retain some of these
+                objects.
+            index_mpid (bool): the same as in get_data
+            line_mode (bool): the bandstructure returned in line mode (True) or
```

Comment:
I would remove this ``line_mode`` argument, it is too specific to a single data type.

Maybe just have the properties be "bandstructure" for uniform and "bandstructure_line" or "bandstructure:line" or something similar for line-mode.